### ISSUE TITLE
Number of entries not properly 'merged'

### ISSUE THREAD

**Issue by peremato** [Moelf/FHist.jl#101]
If you merge! two histograms the number of entries are not accumulated.
```Julia
julia> h1 = Hist1D([1.,2.,3.])
edges: 1.0:1.0:4.0
bin counts: [1, 1, 1]
total count: 3

julia> nentries(h1)
3

julia> h2 = Hist1D([1.,2.,3.])
edges: 1.0:1.0:4.0
bin counts: [1, 1, 1]
total count: 3

julia> nentries(h2)
3

julia> merge!(h1,h2)
edges: 1.0:1.0:4.0
bin counts: [2, 2, 2]
total count: 6

julia> nentries(h1)
3
```

**Comment by tamasgal** [comment]
Fix is on the way in #102 but there is some racing condition I need to understand yet.

**Comment by peremato** [comment]
Thanks @tamasgal!  
Since we are on this, I am getting these warnings when accumulating (!merge) histograms that have been using weights during filling (push!). 
```
┌ Warning: Weights was probably used but StatsBase.Histogram doesn't record # of entries
└ @ FHist [/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35](https://file+.vscode-resource.vscode-cdn.net/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35)
┌ Warning: Weights was probably used but StatsBase.Histogram doesn't record # of entries
└ @ FHist [/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35](https://file+.vscode-resource.vscode-cdn.net/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35)
┌ Warning: Weights was probably used but StatsBase.Histogram doesn't record # of entries
└ @ FHist [/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35](https://file+.vscode-resource.vscode-cdn.net/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35)
┌ Warning: Weights was probably used but StatsBase.Histogram doesn't record # of entries
└ @ FHist [/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35](https://file+.vscode-resource.vscode-cdn.net/Users/mato/.julia/packages/FHist/smrpp/src/FHist.jl:35)
```
Does it makes sense?

**Comment by Moelf** [comment]
this happens when you make a FHist histogram out of a `h::StatsBase histogram` but the `sum(h.weights`) is not integer valued, this means that the `nentries` is not equal to the integral of the `h.weights` but we can't deduce the correct nentries because it's lost information

**Comment by Moelf** [comment]
oh yeah, the fact that `merge!()` triggers this warning is NOT intended.

**Comment by Moelf** [comment]
right, in that case `merge!()` is your only option

**Comment by Moelf** [comment]
this is fixed in 0.11
```julia
julia> h1 = Hist1D([1.,2.,3.])
edges: [1.0, 2.0, 3.0, 4.0]
bin counts: [1.0, 1.0, 1.0]
total count: 3.0

julia> nentries(h1)
3

julia> h2 = Hist1D([1.,2.,3.])
edges: [1.0, 2.0, 3.0, 4.0]
bin counts: [1.0, 1.0, 1.0]
total count: 3.0

julia> nentries(h2)
3

julia> merge!(h1,h2)
edges: [1.0, 2.0, 3.0, 4.0]
bin counts: [2.0, 2.0, 2.0]
total count: 6.0

julia> nentries(h1)
6
```
